Gems/ROS2/Code/Source/ProximitySensor/ROS2ProximitySensor.cpp

```diff
@@ -0,0 +1,141 @@
+        AZStd::string fullTopic = ROS2Names::GetNamespacedName(GetNamespace(), publisherConfig.m_topic);
+        m_detectionPublisher = ros2Node->create_publisher<std_msgs::msg::Bool>(fullTopic.data(), publisherConfig.GetQoS());
+
+        if (m_sensorConfiguration.m_visualise)
+        {
+            auto* entityScene = AZ::RPI::Scene::GetSceneForEntityId(GetEntityId());
+            m_drawQueue = AZ::RPI::AuxGeomFeatureProcessorInterface::GetDrawQueueForScene(entityScene);
+        }
+
+        AZ::Transform entityTransform;
+        AZ::TransformBus::EventResult(entityTransform, GetEntityId(), &AZ::TransformBus::Events::GetWorldTM);
+
+        m_detectionDirection = entityTransform.TransformVector(AZ::Vector3::CreateAxisX());
+        m_detectionDirection.Normalize();
+
+        ROS2SensorComponent::Activate();
+    }
+
+    void ROS2ProximitySensor::Deactivate()
+    {
+        m_detectionPublisher.reset();
+        ROS2SensorComponent::Deactivate();
+    }
+
+    void ROS2ProximitySensor::Visualise()
+    {
+        if (m_drawQueue)
+        {
+            AZStd::vector<AZ::Vector3> linePoints;
+
+            AZ::Vector3 entityTranslation;
+            AZ::TransformBus::EventResult(entityTranslation, GetEntityId(), &AZ::TransformBus::Events::GetWorldTranslation);
+
+            linePoints.push_back(entityTranslation);
+            linePoints.push_back(entityTranslation + m_detectionDirection * m_detectionDistance);
+
+            const uint8_t pixelSize = 5;
+            AZ::RPI::AuxGeomDraw::AuxGeomDynamicDrawArguments drawArgs;
+            drawArgs.m_verts = linePoints.data();
+            drawArgs.m_vertCount = linePoints.size();
+            drawArgs.m_colors = m_wasObjectDetected ? &AZ::Colors::Green : &AZ::Colors::Red;
+            drawArgs.m_colorCount = 1;
+            drawArgs.m_opacityType = AZ::RPI::AuxGeomDraw::OpacityType::Opaque;
+            drawArgs.m_size = pixelSize;
+            m_drawQueue->DrawLines(drawArgs);
+        }
+    }
+
+    void ROS2ProximitySensor::FrequencyTick()
+    {
+        auto* sceneInterface = AZ::Interface<AzPhysics::SceneInterface>::Get();
+
+        AZ::Vector3 entityTranslation;
+        AZ::TransformBus::EventResult(entityTranslation, GetEntityId(), &AZ::TransformBus::Events::GetWorldTranslation);
+
+        AzPhysics::RayCastRequest request;
+        request.m_start = entityTranslation;
+        request.m_direction = m_detectionDirection;
+        request.m_distance = m_detectionDistance;
+
+        if (AzPhysics::SceneHandle sceneHandle = sceneInterface->GetSceneHandle(AzPhysics::DefaultPhysicsSceneName);
+            sceneHandle != AzPhysics::InvalidSceneHandle)
+        {
+            AzPhysics::SceneQueryHits result = sceneInterface->QueryScene(sceneHandle, &request);
+
+            std_msgs::msg::Bool msg;
+            m_wasObjectDetected = !result.m_hits.empty();
+            msg.set__data(m_wasObjectDetected);
```

Comment:
These setters return the structure itself, which here is discarded. They are meant for setter chaining. Here, just use msg.data =